### DISCUSSION TITLE
Where shall I keep my Engineering Notebook?

### DISCUSSION THREAD

**Discussion by edreamleo** [python-rope/rope#599] [Developer]
Over the years I have made dozens (hundreds?) of ENB (Engineering Notebook) entries in [Leo's forum](https://groups.google.com/g/leo-editor). These ENBs are a permanent, public part of the Leonine world. ENBs have recorded the evolution of many breakthroughs in Leo's design and code. ENBs have often been pre-writing for documentation.

As discussed previously, I have just become aware of the depth of my ignorance concerning what Rope does and how it works :-) A series of ENBs seems inevitable.

But where to put these ENBs?

- ekr-study-rope's discussions won't be permanent.
- My personal blog won't do: I want GitHub's fonts, etc.

@lieryan Would it be alright with you for me to create ENBs here? There might be many of them.

**Comment by edreamleo** [comment]
I've just discovered [Using Rope As A Library](https://rope.readthedocs.io/en/latest/library.html#using-rope-as-a-library). Yes, that page will be a useful overview. Otoh, I am mostly interested in understanding *exactly* how soi works.

**Comment by lieryan** [comment]
Rope currently has two documentations. 

The first one is the one you've discovered, in the /docs folder in the source code is the oldest and rope's primary documentation. It's automatically published to [RTD
](http://rope.readthedocs.io/) with every change to master.

The second one is the [Github Wiki](https://github.com/python-rope/rope/wiki), which is basically just a Markdown repository (you can git clone it to grab a copy for local editing). This is where I've been putting new docs and notes that doesn't really fit the existing /docs folder, it's a bit of a distance away from the codebase as there are sometimes things it contains that are only somewhat related to rope. 

Finally, there's also a cache of hidden third "documentation" in the git history. This is mostly the implementation notes by the guy who originally wrote rope. They often contain numerous insights of what he was thinking when he developed a feature. Problem is that they're kinda scattered, since he had a habit of deleting the notes from the codebase once he's done implementing a feature and he no longer need those notes himself, so there's no single checkout that contained all those notes.

**Comment by edreamleo** [comment]
@lieryan Thanks for the links. The Wiki seems good to me.

Do you know the github name of the person who wrote the hidden notes? If so, git should be able to filter their commits.

**Reply by lieryan** [reply]
[aligrudi](https://github.com/python-rope/rope/commits?author=aligrudi) is the original author of Rope.

**Comment by edreamleo** [comment]
Thanks. I'll look through the history to see if I can create an index of aligrudi's comments.

**Comment by edreamleo** [comment]
@lieryan A quick look at `git log --author=aligrudi` doesn't reveal much. I'll try other ways of studying the code.
